src/txn/txn.c

```diff
@@ -529,9 +529,14 @@ __wt_txn_config(WT_SESSION_IMPL *session, const char *cfg[])
 
 	/* Check if prepared updates should be ignored during reads. */
 	WT_RET(__wt_config_gets_def(session, cfg, "ignore_prepare", 0, &cval));
-	if (cval.val ||
-	    (cval.len > 0 && WT_STRING_MATCH("force", cval.str, cval.len)))
-		F_SET(txn, WT_TXN_IGNORE_PREPARE);
+	if (cval.val)
+	{
+		if (cval.len > 0 &&
+		    WT_STRING_MATCH("force", cval.str, cval.len))
+			F_SET(txn, WT_TXN_IGNORE_PREPARE);
+		else
+			F_SET(txn, WT_TXN_IGNORE_PREPARE | WT_TXN_READONLY);
+	}
```

Comment:
Suggest losing the top `if (cval.val)` and replacing with `if (cval.len > 0 && WT_STRING_MATCH("force", cval.str, cval.len) ... else if (cval.val) ...`